configuration-management.yaml

```diff
@@ -0,0 +1,302 @@
+                ConfigurationExample:
+                  $ref: '#/components/examples/ConfigurationExample'
+    put:
+      tags:
+      - "Configuration Management"
+      summary: "Send the new Configuration to apply to the system"
+      description: "Send the new Configuration that will serve as the new system configuration. Can only be invoked by admins."
+      operationId: "updateConfiguration"
+      security:
+        - uc4_token_login: []
+        - uc4_token_login_header: []
+      requestBody:
+        description: "Configuration object that will serve as the new system configuration"
+        required: true
+        content:
+          application/json:
+            schema:
+              $ref: '#/components/schemas/Configuration'
+            examples:
+              ConfigurationExample:
+                $ref: '#/components/examples/ConfigurationExample'
+      responses:
+        "200":
+          description: "OK"
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/Configuration'
+        "400":
+          description: | 
+            Bad Request  
+            types: MalformedLoginToken, Deserialization, MultipleAuthorizationError, JsonValidation
+          content:
+            application/json:
+              schema:
+                oneOf:
+                  - $ref: '#/components/schemas/GenericError'
+                  - $ref: '#/components/schemas/DetailedError'
+        "401":
+          description: |
+            Unauthorized  
+            types: JwtAuthorization, LoginTokenExpired
+          content:
+            application/json:  
+              schema:
+                $ref: '#/components/schemas/GenericError'
+        "403":
+          description: |
+            Forbidden  
+            types: NotEnoughPrivileges
+          content:
+            application/json:  
+              schema:
+                $ref: '#/components/schemas/GenericError'
+        "422":
+          description: |
+            Unprocessable Entity  
+            types: LoginTokenSignatureInvalid
+          content:
+            application/json: 
+              schema:
+                $ref: '#/components/schemas/GenericError'
+  /semester:
+    get:
+      tags:
+      - "Configuration Management"
+      summary: "Get the semester of the given date"
+      description: "Get which semester corresponds to the given date"
+      operationId: "getSemester"
+      parameters:
+      - name: "date"
+        in: "query"
+        description: "Date for which the semester needs to be fetched"
+        required: false
+        schema:
+          type: 'string'
+      responses:
+        "200":
+          description: "OK"
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/JsonSemester'
+              examples:
+                ConfigurationExample:
+                  $ref: '#/components/examples/JsonSemester'
```

Comment:
Please add an example query (date format is not part of this documentation yet)

Reply:
Done